test/jdk/javax/swing/JPopupMenu/bug5091257.java

```diff
@@ -0,0 +1,151 @@
+                            isKeyOk = true;
+                        }
+                    }
+
+                    public void keyPressed(KeyEvent e) {
+                        if (e.getKeyCode() == KeyEvent.VK_CONTEXT_MENU) {
+                            isKeyOk = true;
+                        }
+                    }
+
+                    public void keyReleased(KeyEvent e) {
+                        if (e.getKeyCode() == KeyEvent.VK_CONTEXT_MENU) {
+                            isKeyOk = true;
+                        }
+                    }
+                });
+                mainFrame = new JFrame("Bug5091257");
+                mainFrame.setLayout(new BorderLayout());
+                mainFrame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+                mainFrame.add(button, BorderLayout.CENTER);
+                mainFrame.pack();
+                mainFrame.setVisible(true);
+            });
+            robot.waitForIdle();
+            robot.delay(1000);
+
+            SwingUtilities.invokeAndWait(() -> {
+                loc = button.getLocationOnScreen();
+                loc.x = loc.x + (int) (button.getWidth() / 2);
+                loc.y = loc.y + (int) (button.getHeight() / 2);
+            });
+            robot.mouseMove(loc.x, loc.y);
+            robot.mousePress(MouseEvent.BUTTON3_DOWN_MASK);
+            robot.mouseRelease(MouseEvent.BUTTON3_DOWN_MASK);
+
+            robot.waitForIdle();
+            robot.delay(100);
+
+            try {
+                robot.keyPress(KeyEvent.VK_CONTEXT_MENU);
+                robot.setAutoDelay(10);
+                robot.keyRelease(KeyEvent.VK_CONTEXT_MENU);
+            } catch (IllegalArgumentException ex) {
+                isKeyOk = false;
+            }
+
+            if (!isKeyOk) {
+                System.out.println("KeyEvent can't create or deliver " +
+                        "VK_CONTEXT_MENU event to component. Testing skipped.");
+                passed = true;
+            } else {
+                SwingUtilities.invokeAndWait(() -> {
+                    popup = new JPopupMenu();
+                    popup.add("Item to make popup not empty");
+                    popup.addPopupMenuListener(new PopupMenuListener() {
+                        public void popupMenuWillBecomeVisible(PopupMenuEvent e) {
+                            System.out.println("Popup menu became visible " +
+                                    "on context menu key press. Test passed.");
+                            passed = true;
+                        }
+
+                        public void popupMenuWillBecomeInvisible(PopupMenuEvent e) {
+                        }
+
+                        public void popupMenuCanceled(PopupMenuEvent e) {
+                        }
+                    });
+                    button.setComponentPopupMenu(popup);
+                });
+                robot.keyPress(KeyEvent.VK_CONTEXT_MENU);
+                robot.setAutoDelay(10);
```

Comment:
should be delay...autoDelay should be set while creating robot instance...also if you set autoDelay there's no need of additional delay between key events..